engine/classes/Elgg/DataStorageService.php

```diff
@@ -0,0 +1,40 @@
+<?php
+namespace Elgg;
+use Elgg\Filesystem\Adapter\Adapter;
+
+class DataStorageService {
+	private $logger;
+	private $adapter;
+	
+	// @todo file/dir object cache.
+	
+	
+	public function __construct(Logger $logger) {
+		$this->logger = $logger;
+	}
+	
+	public function setAdapter(Adapter $adapter) {
```

Comment:
> I'd recommend just using constructor params instead of setter
Services don't support constructor arguments.

> Actually, I'm not sure I understand what this class is for. The API is no different than the adapters, really.
Hmm...I don't remember why I separated these out now, either. I'll double check my notes on Monday at work, but I agree the two APIs can probably be combined.